HOTFIX: Handle pagination for SocrataApi so all data is fetched for Marin

benghancock opened the pull request:
This is a hotfix to implement pagination when fetching data via the `resource()` method of the `SocrataApi()` class. The absence of pagination limits the number of records fetched for a given county, which is currently noticeable for Marin on the front-end.

Fixes #203

Mr0grog commented on covid19_sfbayarea/data/utils.py:
```diff
@@ -48,8 +53,37 @@ def request(self, url: str, params: Dict = None, **kwargs: Any) -> Dict:
+    def resource(self, resource_id: str, **kwargs: Any) -> List[Dict]:
+        data: List[Dict] = []
+
+        if "params" in kwargs:
+            if "$offset" not in kwargs["params"]:
+                kwargs["params"].update({"$offset": 0})
+
+            else:
+                pass
+
+        else:
+            kwargs["params"] = {"$offset": 0}
```
I think you could simplify the logic here if you pulled `params` out off `kwargs`. i.e. make the method signature:

```py
def resource(self, resource_id: str, params: dict = None, **kwargs: Any) -> List[Dict]:
```

And then the logic here can be:

```py
    params = params or {}
    params.setdefault("$offset", 0)
    params.setdefault("$limit", self.DEFAULT_LIMIT)

    # ...and down where you make the request:
    results = self.request(f'{self.resource_url}{resource_id}', params=params, **kwargs)
```

As a side note: I think you should explicitly set the `$limit` query param here; we shouldn’t rely on all Socrata servers using the same default or that future updates will keep this default.

Mr0grog reviewed (approved):
This looks great. I left a couple notes on improvements inline, but I don’t think they’re anything critical.

Another thing that might be good to do while you’re messing with this code would be to move this class to its own module file, like the other client classes (`ArcGisFeatureServer`, `Ckan`, `QlikClient`).

benghancock commented:
Thanks for the helpful suggestions @Mr0grog! I've taken up your recommendations and simplified how we set the default `$limit`, ensuring that we're always using either that default value or the value provided as a param when paging. I've also moved this code to its own module, updating the import statements in all the other county scraper modules that use the `SocrataApi` class (Marin, San Francisco, and Santa Clara). If you wouldn't mind taking a final pass, that would be great!

Mr0grog reviewed (approved):
Nice! Merge away. :shipit:
